### ISSUE TITLE
Shipment state should not be based on hidden stock for on demand products

### ISSUE THREAD

**Issue by RachL** [openfoodfoundation/openfoodnetwork#4306]
## Description
Currently on demand products which have a hidden stock at "0" 

## Steps to Reproduce
1. Order a product that is "on demand" and make sure the "hidden" stock value is at 0
2. Go to `/orders?q[s]=completed_at+desc`
3.  See that your order has a shipment state as "out of stock" or "backordered"
4. Go back to the product, uncheck the "on demand checkbox", add value other than 0 and save
5. Now your same order has a shipment state of "pending" or "ready"

## Animated Gif/Screenshot

Step 1 :

![image](https://user-images.githubusercontent.com/6525576/65444747-fd3de800-de30-11e9-93b9-295f20c435a1.png)

After I've added some hidden stock on the product:

![image](https://user-images.githubusercontent.com/6525576/65444936-69b8e700-de31-11e9-9f55-145cef47caa4.png)

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment
* Version used: v2.5
* Browser name and version: Firefox 69
* Operating System and version (desktop or mobile): Both

**Comment by luisramos0** [comment]
this is related to #4267

**Comment by Matt-Yorkley** [comment]
I don't think this issue is valid anymore, right? Is it still possible to get the order state to change to `backordered`?
